testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeRequireQueryPartitionsFilter.java

```diff
@@ -0,0 +1,95 @@
+                "AS VALUES (1, 11), (2, 22) AS data(x, part)");
+
+        try {
+            assertThat(onTrino().executeQuery("SELECT * FROM " + tableName)).containsOnly(row(1, 11), row(2, 22));
+            assertThat(onDelta().executeQuery("SELECT * FROM " + tableName)).containsOnly(row(1, 11), row(2, 22));
+
+            onTrino().executeQuery("SET SESSION delta.query_partition_filter_required = true");
+
+            assertQueryFailure(() -> onTrino().executeQuery("SELECT COUNT(*) FROM " + tableName))
+                    .hasMessageMatching(format("Query failed \\(#\\w+\\): Filter required on default\\.%s for at least one partition column: part", tableName));
+            assertThat(onDelta().executeQuery("SELECT * FROM " + tableName)).containsOnly(row(1, 11), row(2, 22));
+
+            assertThat(onTrino().executeQuery(format("SELECT * FROM %s WHERE part = 11", tableName))).containsOnly(row(1, 11));
+            assertThat(onDelta().executeQuery(format("SELECT * FROM %s WHERE part = 11", tableName))).containsOnly(row(1, 11));
+        }
+        finally {
+            dropDeltaTableWithRetry(tableName);
+        }
+    }
+
+    @Test(groups = {DELTA_LAKE_DATABRICKS, DELTA_LAKE_OSS, PROFILE_SPECIFIC_TESTS})
+    public void testRequiresQueryPartitionFilterWithUppercaseColumnName()
+    {
+        String tableName = "test_require_partition_filter_" + randomNameSuffix();
+
+        onDelta().executeQuery("" +
+                "CREATE TABLE default." + tableName + " " +
+                 " (X INT, PART INT) " +
+                "USING DELTA " +
+                "PARTITIONED BY (PART) " +
+                "LOCATION 's3://" + bucketName + "/databricks-compatibility-test-" + tableName + "' ");
+
+        try {
+            onDelta().executeQuery("INSERT INTO default." + tableName + " VALUES (1, 11), (2, 22)");
+
+            assertThat(onTrino().executeQuery("SELECT * FROM " + tableName)).containsOnly(row(1, 11), row(2, 22));
+            assertThat(onDelta().executeQuery("SELECT * FROM " + tableName)).containsOnly(row(1, 11), row(2, 22));
+
+            onTrino().executeQuery("SET SESSION delta.query_partition_filter_required = true");
+
+            assertThat(onTrino().executeQuery(format("SELECT * FROM %s WHERE part = 11", tableName))).containsOnly(row(1, 11));
+            assertThat(onTrino().executeQuery(format("SELECT * FROM %s WHERE PART = 11", tableName))).containsOnly(row(1, 11));
+            assertThat(onTrino().executeQuery(format("SELECT * FROM %s WHERE Part = 11", tableName))).containsOnly(row(1, 11));
+
+            assertThat(onDelta().executeQuery(format("SELECT * FROM %s WHERE part = 11", tableName))).containsOnly(row(1, 11));
+            assertThat(onDelta().executeQuery(format("SELECT * FROM %s WHERE PART = 11", tableName))).containsOnly(row(1, 11));
+            assertThat(onDelta().executeQuery(format("SELECT * FROM %s WHERE Part = 11", tableName))).containsOnly(row(1, 11));
```

Comment:
Remove these assertions. They aren't helpful because `delta.query_partition_filter_required` obviously doesn't affect to Delta query execution.